### ISSUE TITLE
Build failure with Rust 1.80.0

### ISSUE THREAD

**Issue by bowlofeggs** [ClementTsang/bottom#1500]
### Checklist

- [X] I've looked through the [troubleshooting docs](https://clementtsang.github.io/bottom/nightly/troubleshooting), [the known problems list](https://clementtsang.github.io/bottom/nightly/support/official/#known-problems), and  [existing open issues](https://github.com/ClementTsang/bottom/issues?q=is%3Aopen+is%3Aissue) for similar issues.


### What operating system and version are you using?

Gentoo Linux

### What architecture are you using?

x86_64/AMD64

### What terminal(s) are you running bottom on that are experiencing the problem?

_No response_

### What filesystem(s) are you using?

Btrfs

### What version of bottom are you running?

0.9.6

### How did you install bottom?

It will fail if installed with Gentoo (which builds it), or when using `cargo build` with Rust 1.80.0.

### Describe the issue

Bottom uses an older version of `time`, which leads to a build failure with Rust 1.80.0.

### What is the expected behaviour?

It should build successfully.

### What is the actual behaviour?

```
error[E0282]: type annotations needed for `Box<_>`                                                                                                                                                                                                                     23:37:42 [492/1289]
  --> /var/tmp/portage/sys-process/bottom-0.9.6/work/cargo_home/gentoo/time-0.3.27/src/format_description/parse/mod.rs:83:9
   |                                                                  
83 |     let items = format_items                                                                                                                                                                                                                                                         
   |         ^^^^^                                                                                                                                                                                                                                                                        
...                                                                                                                                                                                                                                                                                       
86 |     Ok(items.into())                                                                                                                                                                                                                                                                 
   |              ---- type must be known at this point                                                                                                                                                                                                                                   
   |                                                                                                                                                                                                                                                                                      
help: consider giving `items` an explicit type, where the placeholders `_` are specified                                                     
   |                                                                                                                                         
83 |     let items: Box<_> = format_items                                                                                                    
   |              ++++++++                                                                                                                   
                                                                                                                                             
For more information about this error, try `rustc --explain E0282`.
error: could not compile `time` (lib) due to 1 previous error
```

### How can we reproduce this?

Build bottom with Rust 1.80.0.

### Additional information

_No response_

**Comment by bowlofeggs** [comment]
It looks like this is fixed on `main`, so we just need a release to get it fixed for end users who `cargo install`.

**Comment by bowlofeggs** [comment]
See https://github.com/rust-lang/rust/issues/127343

**Comment by ClementTsang** [comment]
Darn, was hoping I wouldn't be affected by this, but thanks for the heads up. I'll do a release in a bit, I'll probably just cherry-pick the change from `main` as needed.

**Comment by ClementTsang** [comment]
Marking this as closed, published 0.9.7.
